api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java

```diff
@@ -54,10 +54,21 @@ public boolean isEmpty() {
     return wrapperSet.isEmpty();
   }
 
+  public CharSequenceWrapper wrapper() {
+    if (containsWrapper == null) {
+      synchronized (this) {
+        if (containsWrapper == null) {
+          this.containsWrapper = ThreadLocal.withInitial(() -> CharSequenceWrapper.wrap(null));
```

Comment:
Wondering if making sense to have a static ThreadLocal object shared by all `CharSequenceSet`?

Reply:
Yes, this is a simpler solution. Good point!